lib/rspec/core/example_group.rb

```diff
@@ -340,6 +340,16 @@ def self.include_examples(name, *args, &block)
         find_and_eval_shared("examples", name, caller.first, *args, &block)
       end
 
+      # Adds an example to the example group
+      def self.add_example example
+        examples << example
+      end
+
+      # Removes an example from the example group
+      def self.remove_example example
+        examples.delete example
+      end
```

Comment:
Do we want the object ids to be different? I'd assume the example ids should remain the same for the existing examples...